src/vm.rs

```diff
@@ -473,23 +440,51 @@ pub struct RuntimeEnvironment<'a, C: ContextObject> {
 ///
 /// let mut vm = EbpfVm::new(&verified_executable, &mut context_object, memory_mapping, stack_len);
 ///
-/// let (instruction_count, result) = vm.execute_program(true);
+/// let (instruction_count, result) = vm.execute_program(&verified_executable, true);
 /// assert_eq!(instruction_count, 1);
 /// assert_eq!(result.unwrap(), 0);
 /// ```
+#[repr(C)]
 pub struct EbpfVm<'a, V: Verifier, C: ContextObject> {
-    pub(crate) executable: &'a Executable<V, C>,
+    /// Needed to exit from the guest back into the host
+    pub host_stack_pointer: *mut u64,
+    /// The current call depth.
+    ///
+    /// Incremented on calls and decremented on exits. It's used to enforce
+    /// config.max_call_depth and to know when to terminate execution.
+    pub call_depth: u64,
+    /// Guest stack pointer (r11).
+    ///
+    /// The stack pointer isn't exposed as an actual register. Only sub and add
+    /// instructions (typically generated by the LLVM backend) are allowed to
+    /// access it when sbpf_version.dynamic_stack_frames()=true. Its value is only
+    /// stored here and therefore the register is not tracked in REGISTER_MAP.
+    pub stack_pointer: u64,
+    /// Pointer to ContextObject
+    pub context_object_pointer: &'a mut C,
+    /// Last return value of instruction_meter.get_remaining()
+    pub previous_instruction_meter: u64,
+    /// CPU cycles accumulated by the stop watch
+    pub stopwatch_numerator: u64,
+    /// Number of times the stop watch was used
+    pub stopwatch_denominator: u64,
+    /// ProgramResult inlined
+    pub program_result: ProgramResult,
+    /// MemoryMapping inlined
+    pub memory_mapping: MemoryMapping<'a>,
+    /// Stack of CallFrames used by the Interpreter
+    pub call_frames: Vec<CallFrame>,
     /// TCP port for the debugger interface
     #[cfg(feature = "debugger")]
     pub debug_port: Option<u16>,
-    /// Runtime state
-    pub env: RuntimeEnvironment<'a, C>,
+    /// Programs must have passed this verifier to be executed
+    _verifier: PhantomData<V>,
```

Comment:
Can't we remove this, move the `V` bound to `new` and then it becomes impossible
to create `EbpfVm` for a non-verified program? Then we can remove one generic
argument from `EbpfVm`